repose-aggregator/components/filters/uri-stripper-filter/src/main/scala/org/openrepose/filters/uristripper/UriStripperFilter.scala

```diff
@@ -102,7 +105,30 @@ class UriStripperFilter @Inject()(configurationService: ConfigurationService)
         wrappedRequest.setRequestURI(joinTokensIntoPath(uriTokens))
       }
 
-      filterChain.doFilter(wrappedRequest, wrappedResponse)
+      val requestLinkPaths = config.getLinkResource
+        .filter(resource => resource.getUriPathRegex =~ wrappedRequest.getRequestURI)
+        .filter(resource => isMatchingMethod(resource.getHttpMethods.toSet, wrappedRequest.getMethod))
+        .map(resource => getPathsForContentType(wrappedRequest.getContentType, resource.getRequest))
+        .fold(List.empty[LinkPath])(_ ++ _)
```

Comment:
It looks like that last line compresses a `List[List[LinkPath]]` down to a `List[LinkPath]`.

The better way to do this would probably be to change the `map` line to a `flatMap` and then call `toList` at the end.